docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,577 @@
+
+### Short names
+
+Make command names as short and easy to spell as possible. For example, if `class` is clear enough don't make the command `classification`.
+
+### Lowercase names
+
+Define names in lowercase only, except you can make uppercase aliases to make commands or options case insensitive.
+
+### Kebab case names
+
+Use [kebab case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) to distinguish words. For example, the .NET CLI option that is defined as [`--additionalprobingpath`](../../core/tools/dotnet.md#runtime-options) should have been defined as `--additional-probing-path`.
+
+Case sensitivity does not apply to options that are based on enums. Enum names are matched regardless of casing.
+
+### Pluralization
+
+Within an app, be consistent in pluralization. For example, don't mix plural and singular names for options that can have multiple values (maximum arity greater than one):
+
+| Option names                                 | Consistency  |
+|----------------------------------------------|--------------|
+| `--additional-probing-paths` and `--sources` | ✔️          |
+| `--additional-probing-path` and `--source`   | ✔️          |
+| `--additional-probing-paths` and `--source`  | ❌          |
+| `--additional-probing-path` and `--sources`  | ❌          |
+
+### Verbs vs. nouns
+
+Use verbs rather than nouns for commands that refer to actions (those without subcommands under them), for example: `dotnet workload remove`, not `dotnet workload removal`. And use nouns rather than verbs for options, for example: `--configuration`, not `--configure`.
+
+### The `--verbosity` option
+
+`System.CommandLine` applications typically offer a `--verbosity` option that specifies how much output is sent to the console. Here are the standard five settings:
+
+* `Q[uiet]`
+* `M[inimal]`
+* `N[ormal]`
+* `D[etailed]`
+* `Diag[nostic]`
+
+These are the standard names, but existing apps sometimes use `Silent` in place of `Quiet`, and `Trace`, `Debug`, or `Verbose` in place of `Diagnostic`.
+
+Each app defines its own criteria that determine what gets displayed at each level. Typically an app only needs three levels:
+
+* Quiet
+* Normal
+* Diagnostic
+
+If an app doesn't need five different levels, the option should still define the same five settings. In that case, `Minimal` and `Normal` will produce the same output, and `Detailed` and `Diagnostic` will likewise be the same.
+
+The expectation for `Quiet` is that no output is displayed on the console. However, if an app offers an interactive mode, the app should do one of the following alternatives:
+
+* Display prompts for input when `--interactive` is specified, even if `--verbosity` is `Quiet`.
+* Disallow the use of `--verbosity Quiet` and `--interactive` together.
+
+Otherwise the app will wait for input without telling the user what it's waiting for.
```

Comment:
Maybe add something like: 

```suggestion
Otherwise the app will wait for input without telling the user what it's waiting for. It will appear that your application froze and the user will have no idea the application is waiting for input. 
```